### PR TITLE
[Feature] - Add checkoutByTag API endpoint for assets

### PR DESCRIPTION
**Pull request by ntbutler-nbcs**
# Description

Seeing as asset tags are unique, it should be safe to, and would be useful to be able to call the API and check out an asset using it's asset tag. We have each asset tagged with a barcode, and have an internally developed mobile app to quickly scan user ID cards, then the asset tag. 

This extra endpoint means we no longer need to run a separate API call to look up the asset to get it's internal snipe ID first.

This uses essentially the same code from the GET /hardware/bytag/:asset_tag endpoint, and if the asset is found, calls the existing checkout function.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update
**NOTE** - there seems to be an undocumented new API endpoint for checkinByTag, which should be updated too. 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test API call with existing deployable check-in asset - asset checked out as expected
- [x] Test API call with existing deployable check-out asset - asset not available to be checked out as expected
- [x] Test API call with non-existing asset tag - asset not found as expected

**Test Configuration**:
* PHP version: 8.0.20
* MySQL version: 8.0
* Webserver version: 6.0.2
* OS version: Windows Server 2019

# Checklist:

- [x] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [x] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Http/Controllers/Api/AssetsController.php [hunk 1]
```diff
@@ -746,7 +746,21 @@ public function restore(Request $request, $assetId = null)
...
+     * Checkout an asset by its tag.
+     *
+     * @author [N. Butler]
+     * @param string $tag
+     * @since [v6.0.5]
+     * @return JsonResponse
+     */
+    public function checkoutByTag(AssetCheckoutRequest $request, $tag)
+    {
+        if ($asset = Asset::with('assetstatus')->with('assignedTo')->where('asset_tag', $tag)->first()) {
```
Comment (snipe) [inline app/Http/Controllers/Api/AssetsController.php]: Do we actually need the `with('assetstatus')->with('assignedTo')` stuff? The `checkout()` method already tries to determine if we can check the item out or not.

File: config/version.php [hunk 2]
```diff
@@ -1,10 +1,10 @@
 <?php
 return array (
   'app_version' => 'v6.0.5',
-  'full_app_version' => 'v6.0.5 - build 8229-g94e723a88',
```
Comment (snipe) [inline config/version.php]: These changes aren't correct - can you revert this file?

**Review by snipe — CHANGES_REQUESTED** [review]
Some small changes requested

File: app/Http/Controllers/Api/AssetsController.php (hunk 1 above)
Comment (ntbutler-nbcs) [reply app/Http/Controllers/Api/AssetsController.php]: Absolutely - my bad, I did remove it but at one point, but must have added it back when I was working on my own checkinByTag function (before I found there already was one). Changes applied.

File: config/version.php (hunk 2 above)
Comment (ntbutler-nbcs) [reply config/version.php]: Apologies for that, first-timer on pull requests - must have had the master version. File reverted to dev version.

**Pull request merged**